src/core/Database.cpp

```diff
@@ -355,3 +320,34 @@ const CompositeKey & Database::key() const
     return m_data.key;
 }
 
+QVariantMap& Database::kdfParams() {
+    return m_data.kdfParams;
+}
+
+void Database::setKdfParams(QVariantMap params) {
```

Comment:
As long as the class does not contain any other private fields, the size will be the same as the size of a pointer. BUT: as a user of the class you don't know that. For most Qt classes it may be true, but it's not generally true, whereas const references ALWAYS guarantee that you only pass around 8 byte. If you just pass everything by const reference (except smart pointers), you don't need to worry about whether a class does implicit sharing or not. Most of our own classes don't.

const references also make your code cleaner, because you need to be clear about when you want to modify an object and when not. A big disadvantage of implicit sharing is also that it is very easy to accidentally detach data (e.g. by using a non-const container in a C++11 range-based for loop). If you pass it by const reference, you don't have that issue (because the container is already const).